document/src/vespa/document/fieldvalue/mapfieldvalue.cpp

```diff
@@ -266,10 +313,9 @@ MapFieldValue::find(const FieldValue& key) const
 MapFieldValue::iterator
 MapFieldValue::find(const FieldValue& key)
 {
-    size_t sz = _keys->size();
-    if ((sz > 0) && (key.getClass().id() == (*_keys)[0].getClass().id())) {
-        for (size_t i(0), m(_keys->size()); i < m; i++) {
-            if ((*_keys)[i].fastCompare(key) == 0) {
+    if ((size() > 0) && (key.getClass().id() == (*_keys)[0].getClass().id())) {
```

Comment:
Just to make sure: since erasing elements only tags the element as not present, this means that if `size() > 0`, `(*_keys)[0]` is guaranteed to be non-null even if `_present[0] == false`, correct?